runtime/drivers/druid/druid_api_sql_driver.go

```diff
@@ -0,0 +1,354 @@
+package druid
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type druidSQLDriver struct{}
+
+var _ driver.Driver = &druidSQLDriver{}
+
+func (a *druidSQLDriver) Open(dsn string) (driver.Conn, error) {
+	client := http.Client{Timeout: time.Second * 10}
+
+	return &sqlConnection{
+		client: &client,
+		dsn:    dsn,
+	}, nil
+}
+
+func init() {
+	sql.Register("druid", &druidSQLDriver{})
+}
+
+type sqlConnection struct {
+	client *http.Client
+	dsn    string
+}
+
+var _ driver.QueryerContext = &sqlConnection{}
+
+func emptyTransformer(v any) (any, error) {
+	return v, nil
+}
+
+func (c *sqlConnection) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	b, err := json.Marshal(druidRequest(query, args))
+	if err != nil {
+		return nil, err
+	}
+
+	bodyReader := bytes.NewReader(b)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.dsn, bodyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	// nolint:all
```

Comment:
Why the need for `nolint`?

Reply:
we need to carry the body to allow streaming
![image](https://github.com/rilldata/rill/assets/14215045/64b9b0d2-e8fa-4f18-a95c-d0e348b4a0cd)

Reply:
fixed